### ISSUE TITLE
cloud_to_prod and cloud_to_prod_auth interop UNAVAILABLE errors for Go and Java on Kokoro

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#11967]
seems related to https://github.com/grpc/grpc/issues/10763, but on Kokoro

exact test failures seem to be:

* `cloud_to_prod_auth:gateway_v4:javaokhttp:compute_engine_creds`
* `cloud_to_prod:default:javaokhttp:client_streaming`
* `cloud_to_prod:default:javaokhttp:ping_pong`
* `cloud_to_prod_auth:default:go:jwt_token_creds`
* `cloud_to_prod_auth:default:go:compute_engine_creds`
* `cloud_to_prod_auth:gateway_v4:javaokhttp:compute_engine_creds`
* `cloud_to_prod_auth:gateway_v4:java:jwt_token_creds`

```
'./gradlew installDist -PskipCodegen=true' or
'./gradlew :grpc-interop-testing:installDist -PskipCodegen=true' after any
changes. -PskipCodegen=true is optional, but requires less setup.
[jetty-alpn-agent] Using: alpn-boot-8.1.11.v20170118.jar
Running test client_streaming
[jetty-alpn-agent] Replacing: sun/security/ssl/ClientHandshaker
Jul 28, 2017 1:35:18 AM io.grpc.okhttp.internal.Platform getAndroidSecurityProvider
WARNING: Unable to find Conscrypt
[jetty-alpn-agent] Replacing: sun/security/ssl/HelloExtensions
[jetty-alpn-agent] Replacing: sun/security/ssl/ExtensionType
[jetty-alpn-agent] Replacing: sun/security/ssl/Alerts
Jul 28, 2017 1:35:18 AM io.grpc.internal.ChannelExecutor drain
WARNING: Runnable threw exception in ChannelExecutor
java.lang.IllegalStateException: ConnectivityStateManager is already disabled
	at com.google.common.base.Preconditions.checkState(Preconditions.java:174)
	at io.grpc.internal.ConnectivityStateManager.gotoState(ConnectivityStateManager.java:66)
	at io.grpc.internal.ManagedChannelImpl$5.run(ManagedChannelImpl.java:465)
	at io.grpc.internal.ChannelExecutor.drain(ChannelExecutor.java:72)
	at io.grpc.internal.DelayedClientTransport.shutdown(DelayedClientTransport.java:211)
	at io.grpc.internal.ManagedChannelImpl.shutdown(ManagedChannelImpl.java:469)
	at io.grpc.internal.ManagedChannelImpl.shutdown(ManagedChannelImpl.java:68)
	at io.grpc.testing.integration.AbstractInteropTest.tearDown(AbstractInteropTest.java:242)
	at io.grpc.testing.integration.TestServiceClient.tearDown(TestServiceClient.java:170)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:67)

Exception in thread "main" java.lang.RuntimeException: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:185)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:65)
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:476)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:455)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:79)
	at io.grpc.testing.integration.AbstractInteropTest.clientStreaming(AbstractInteropTest.java:394)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:206)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:181)
	... 1 more
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.Status.asRuntimeException(Status.java:526)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:385)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:422)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:61)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:496)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$600(ClientCallImpl.java:425)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:528)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:102)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)
	at io.grpc.okhttp.OkHttpProtocolNegotiator.negotiate(OkHttpProtocolNegotiator.java:88)
	at io.grpc.okhttp.OkHttpTlsUpgrader.upgrade(OkHttpTlsUpgrader.java:63)
	at io.grpc.okhttp.OkHttpClientTransport$1.run(OkHttpClientTransport.java:416)
	... 4 more
Shutting down

2017-07-27 18:35:18,940 FAILED: cloud_to_prod:default:javaokhttp:client_streaming [ret=1, pid=4478]
```

```
0\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000"
}
> but was:<io.grpc.StatusRuntimeException: UNAVAILABLE: End of stream or IOException>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at io.grpc.testing.integration.AbstractInteropTest.pingPong(AbstractInteropTest.java:468)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:214)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:181)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:65)
Shutting down

2017-07-27 18:35:19,210 FAILED: cloud_to_prod:default:javaokhttp:ping_pong [ret=1, pid=4388]
```

```
2017-07-27 18:35:40,740 FATAL: 2017/07/28 01:35:38 /TestService/UnaryCall RPC failed: rpc error: code = Unavailable desc = transport is closing
exit status 1

2017-07-27 18:35:40,741 FAILED: cloud_to_prod_auth:default:go:jwt_token_creds [ret=1, pid=21078]
```

```
2017-07-27 18:35:40,753 FATAL: 2017/07/28 01:35:38 /TestService/UnaryCall RPC failed: rpc error: code = Unavailable desc = transport is closing
exit status 1

2017-07-27 18:35:40,754 FAILED: cloud_to_prod_auth:default:go:compute_engine_creds [ret=1, pid=21077]
```

```
'./gradlew installDist -PskipCodegen=true' or
'./gradlew :grpc-interop-testing:installDist -PskipCodegen=true' after any
changes. -PskipCodegen=true is optional, but requires less setup.
[jetty-alpn-agent] Using: alpn-boot-8.1.11.v20170118.jar
Running test compute_engine_creds
[jetty-alpn-agent] Replacing: sun/security/ssl/ClientHandshaker
Jul 28, 2017 1:35:48 AM io.grpc.okhttp.internal.Platform getAndroidSecurityProvider
WARNING: Unable to find Conscrypt
[jetty-alpn-agent] Replacing: sun/security/ssl/HelloExtensions
[jetty-alpn-agent] Replacing: sun/security/ssl/ExtensionType
[jetty-alpn-agent] Replacing: sun/security/ssl/ClientHandshaker$2
[jetty-alpn-agent] Replacing: sun/security/ssl/Alerts
Jul 28, 2017 1:35:48 AM io.grpc.internal.ChannelExecutor drain
WARNING: Runnable threw exception in ChannelExecutor
java.lang.IllegalStateException: ConnectivityStateManager is already disabled
	at com.google.common.base.Preconditions.checkState(Preconditions.java:174)
	at io.grpc.internal.ConnectivityStateManager.gotoState(ConnectivityStateManager.java:66)
	at io.grpc.internal.ManagedChannelImpl$5.run(ManagedChannelImpl.java:465)
	at io.grpc.internal.ChannelExecutor.drain(ChannelExecutor.java:72)
	at io.grpc.internal.DelayedClientTransport.shutdown(DelayedClientTransport.java:211)
	at io.grpc.internal.ManagedChannelImpl.shutdown(ManagedChannelImpl.java:469)
	at io.grpc.internal.ManagedChannelImpl.shutdown(ManagedChannelImpl.java:68)
	at io.grpc.testing.integration.AbstractInteropTest.tearDown(AbstractInteropTest.java:242)
	at io.grpc.testing.integration.TestServiceClient.tearDown(TestServiceClient.java:170)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:67)

Exception in thread "main" io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:210)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:191)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:124)
	at io.grpc.testing.integration.TestServiceGrpc$TestServiceBlockingStub.unaryCall(TestServiceGrpc.java:473)
	at io.grpc.testing.integration.AbstractInteropTest.computeEngineCreds(AbstractInteropTest.java:1281)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:222)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:181)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:65)
Caused by: java.net.SocketException: Connection reset
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:115)
	at java.net.SocketOutputStream.write(SocketOutputStream.java:155)
	at sun.security.ssl.OutputRecord.writeBuffer(OutputRecord.java:431)
	at sun.security.ssl.OutputRecord.write(OutputRecord.java:417)
	at sun.security.ssl.SSLSocketImpl.writeRecordInternal(SSLSocketImpl.java:876)
	at sun.security.ssl.SSLSocketImpl.writeRecord(SSLSocketImpl.java:847)
	at sun.security.ssl.SSLSocketImpl.writeRecord(SSLSocketImpl.java:717)
	at sun.security.ssl.Handshaker.sendChangeCipherSpec(Handshaker.java:1124)
	at sun.security.ssl.ClientHandshaker.sendChangeCipherAndFinish(ClientHandshaker.java:1269)
	at sun.security.ssl.ClientHandshaker.serverHelloDone(ClientHandshaker.java:1181)
	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:352)
	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1026)
	at sun.security.ssl.Handshaker.process_record(Handshaker.java:961)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)
	at io.grpc.okhttp.OkHttpProtocolNegotiator.negotiate(OkHttpProtocolNegotiator.java:88)
	at io.grpc.okhttp.OkHttpTlsUpgrader.upgrade(OkHttpTlsUpgrader.java:63)
	at io.grpc.okhttp.OkHttpClientTransport$1.run(OkHttpClientTransport.java:416)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:102)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Shutting down

2017-07-27 18:35:48,906 FAILED: cloud_to_prod_auth:gateway_v4:javaokhttp:compute_engine_creds [ret=1, pid=27203]
```

**Comment by adelez** [comment]
Hope adding flake detection (#11975) can help alleviate this.

**Comment by apolcyn** [comment]
`tests.cloud_to_prod:default:javaokhttp:server_streaming` also failed (on jenkins)



failed in (https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/15758/testReport/junit/(root)/tests/cloud_to_prod_default_javaokhttp_server_streaming/)

```
Gradle is no longer run automatically. Make sure to run
'./gradlew installDist -PskipCodegen=true' or
'./gradlew :grpc-interop-testing:installDist -PskipCodegen=true' after any
changes. -PskipCodegen=true is optional, but requires less setup.
[jetty-alpn-agent] Using: alpn-boot-8.1.11.v20170118.jar
Running test server_streaming
[jetty-alpn-agent] Replacing: sun/security/ssl/ClientHandshaker
Jul 29, 2017 2:33:01 AM io.grpc.okhttp.internal.Platform getAndroidSecurityProvider
WARNING: Unable to find Conscrypt
[jetty-alpn-agent] Replacing: sun/security/ssl/HelloExtensions
[jetty-alpn-agent] Replacing: sun/security/ssl/ExtensionType
[jetty-alpn-agent] Replacing: sun/security/ssl/ClientHandshaker$2
[jetty-alpn-agent] Replacing: sun/security/ssl/Alerts
Jul 29, 2017 2:33:01 AM io.grpc.internal.ChannelExecutor drain
WARNING: Runnable threw exception in ChannelExecutor
java.lang.IllegalStateException: ConnectivityStateManager is already disabled
	at com.google.common.base.Preconditions.checkState(Preconditions.java:174)
	at io.grpc.internal.ConnectivityStateManager.gotoState(ConnectivityStateManager.java:66)
	at io.grpc.internal.ManagedChannelImpl$5.run(ManagedChannelImpl.java:465)
	at io.grpc.internal.ChannelExecutor.drain(ChannelExecutor.java:72)
	at io.grpc.internal.DelayedClientTransport.shutdown(DelayedClientTransport.java:211)
	at io.grpc.internal.ManagedChannelImpl.shutdown(ManagedChannelImpl.java:469)
	at io.grpc.internal.ManagedChannelImpl.shutdown(ManagedChannelImpl.java:68)
	at io.grpc.testing.integration.AbstractInteropTest.tearDown(AbstractInteropTest.java:242)
	at io.grpc.testing.integration.TestServiceClient.tearDown(TestServiceClient.java:170)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:67)

Exception in thread "main" java.lang.AssertionError: io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.testing.integration.AbstractInteropTest.assertSuccess(AbstractInteropTest.java:1363)
	at io.grpc.testing.integration.AbstractInteropTest.serverStreaming(AbstractInteropTest.java:360)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:210)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:181)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:65)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.Status.asRuntimeException(Status.java:526)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:385)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:422)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:61)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:496)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$600(ClientCallImpl.java:425)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:528)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:102)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at sun.security.ssl.SSLSocketImpl.waitForClose(SSLSocketImpl.java:1769)
	at sun.security.ssl.HandshakeOutStream.flush(HandshakeOutStream.java:124)
	at sun.security.ssl.Handshaker.sendChangeCipherSpec(Handshaker.java:1130)
	at sun.security.ssl.ClientHandshaker.sendChangeCipherAndFinish(ClientHandshaker.java:1269)
	at sun.security.ssl.ClientHandshaker.serverHelloDone(ClientHandshaker.java:1181)
	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:352)
	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1026)
	at sun.security.ssl.Handshaker.process_record(Handshaker.java:961)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)
	at io.grpc.okhttp.OkHttpProtocolNegotiator.negotiate(OkHttpProtocolNegotiator.java:88)
	at io.grpc.okhttp.OkHttpTlsUpgrader.upgrade(OkHttpTlsUpgrader.java:63)
	at io.grpc.okhttp.OkHttpClientTransport$1.run(OkHttpClientTransport.java:416)
	... 4 more
```

**Comment by adelez** [comment]
Closed as dup of #10763.
